static/src/javascripts/projects/commercial/modules/liveblog-adverts.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+import detect from 'lib/detect';
+import mediator from 'lib/mediator';
+import { addSlot } from 'commercial/modules/dfp/add-slot';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import createSlot from 'commercial/modules/dfp/create-slot';
+import spaceFiller from 'common/modules/article/space-filler';
+
+const INTERVAL = 5; // number of posts between ads
+const OFFSET = 1.5; // ratio of the screen height from which ads are loaded
+const MAX_ADS = 8; // maximum number of ads to display
+
+type spaceFillerRules = {
+    bodySelector: string,
+    slotSelector: string,
+    fromBottom: boolean,
+    startAt: Node | null | typeof undefined, // #? I feel this is circumventing the point of Flow
+    absoluteMinAbove: number,
+    minAbove: number,
+    minBelow: number,
+    filter: (slot: { top: number }, index: number) => boolean,
+};
+
+let SLOTCOUNTER = 0;
+let WINDOWHEIGHT;
+let firstSlot;
+
+const startListening = () => {
+    // eslint-disable-next-line no-use-before-define
+    mediator.on('modules:autoupdate:updates', onUpdate);
+};
+
+const stopListening = () => {
+    // eslint-disable-next-line no-use-before-define
+    mediator.off('modules:autoupdate:updates', onUpdate);
+};
+
+const getWindowHeight = (doc = document): number => {
+    if (doc.documentElement && doc.documentElement.clientHeight) {
+        return doc.documentElement.clientHeight;
+    }
+    return 0; // #? zero, or throw an error?
```

Comment:
Adding a type to doc doesn't work, because I have specified the return type of the function 😅 

I feel that `getWindowHeight` should always return a number, and have the responsibility of deciding what to do if `doc.documentElement.clientHeight` does not exist. After setting the value of WindowHeight, I would rather be passing a zero around, than maybe an undefined.

Flow IDE gives the error: `this type is incompatible with an implicitly-returned undefined` if I don't return a number in the else block.

This will work:
```
const getWindowHeight = (doc = document): number | typeof undefined => {
    if (doc.documentElement && doc.documentElement.clientHeight) {
        return doc.documentElement.clientHeight;
    }
};
```
However, I feel this maligns the spirit of Flow, and I would like to avoid Schrodinger's variable.  Since 0 is falsy, I think it is a better fallback for getting a value than `undefined`, in this particular case.